### PR TITLE
Feat/router hop

### LINKED ISSUE #7334: EPIC: Bridge integration and swap + bridge UI
Add hop integration
design:
https://www.figma.com/file/FkFClTCYKf83RJWoifWgoX/Wallet-v2?node-id=2774%3A143138
https://www.figma.com/file/FkFClTCYKf83RJWoifWgoX/Wallet-v2?node-id=350%3A91908
https://www.figma.com/file/FkFClTCYKf83RJWoifWgoX/Wallet-v2?node-id=2829%3A145815

### PR DESCRIPTION
**Pull request by alaibe**
### What does the PR do
fixes #7334 
Added support for UI for bridge and multi routing transactions

### Affected areas

Wallet: Send 
Browser: Send
ENS and Stickers purchase

### StatusQ checklist

- [ ] add documentation if necessary (new component, new feature)
- [ ] update sandbox app
  - in case of new component, add new component page
  - in case of new features, add variation to existing component page
  - nice to have: add it to the demo application as well
- [ ] test changes in both light and dark theme?

### Screenshot of functionality (including design for comparison)

- [x] I've checked the design and this PR matches it

https://user-images.githubusercontent.com/60327365/196245514-38b85671-80c5-404b-a486-2feaa6f9988e.mov

### CODE REVIEW

File: ui/imports/shared/views/NetworkCardsComponent.qml [hunk 1]
```diff
@@ -127,39 +114,32 @@ Item {
...
-                        store.addRemoveDisabledChain(suggestedRoutes, model.chainId, disabled)
-                        reCalculateSuggestedRoute(store.disabledChainIds)
-                    }
-                    onAdvancedInputTextChanged: {
-                        if(selectedNetwork && selectedNetwork.chainName === model.chainName)
-                            d.customAmountToReceive = isNaN(parseFloat(advancedInputText)) ? 0 : parseFloat(advancedInputText)
+                        store.addRemoveDisabledToChain(model.chainId, disabled)
+                        root.reCalculateSuggestedRoute()
                     }
+//                    onAdvancedInputTextChanged: {
```
Comment (alaibe) [inline ui/imports/shared/views/NetworkCardsComponent.qml:139]: can it be removed?

File: ui/imports/shared/views/NetworkCardsComponent.qml [hunk 2]
```diff
@@ -73,44 +70,34 @@ Item {
...
-                        reCalculateSuggestedRoute(store.disabledChainIds)
-                    }
-                    onAdvancedInputTextChanged: {
-                        if(selectedNetwork && selectedNetwork.chainName === model.chainName) {
-                            d.customAmountToSend = isNaN(parseFloat(advancedInputText)) ? 0 : parseFloat(advancedInputText)
-                        }
+                        store.addRemoveDisabledFromChain(model.chainId, disabled)
+                        root.reCalculateSuggestedRoute()
                     }
+//                    onAdvancedInputTextChanged: {
```
Comment (alaibe) [inline ui/imports/shared/views/NetworkCardsComponent.qml:96]: removed or is it for advance?

File: ui/app/AppLayouts/stores/RootStore.qml [hunk 3]
```diff
@@ -165,8 +172,8 @@ QtObject {
         return walletSectionTransactions.getChainIdForBrowser()
     }
 
-    function suggestedRoutes(account, amount, token, disabledChainIds) {
-        return JSON.parse(walletSectionTransactions.suggestedRoutes(account, amount, token, disabledChainIds)).networks
+    function suggestedRoutes(account, amount, token, disabledFromChainIDs, disabledToChainIDs, preferredChainIds, priority) {
```
Comment (alaibe) [inline ui/app/AppLayouts/stores/RootStore.qml:175]: what do u think of having a transaction store? or send store. I believe this component is so complex it deserves his own store

That way we can remove the need to have sticker store injected into ens view?

File: ui/imports/shared/views/NetworkCardsComponent.qml (hunk 2 above)
Comment (Khushboo-dev-cpp) [reply ui/imports/shared/views/NetworkCardsComponent.qml:96]: it will be needed when we work on custom view

**Review by Khushboo-dev-cpp — APPROVED** [review]

**Comment by jakubgs** [comment]
The build is failing on Windows at checkout with:
```
hudson.plugins.git.GitException: Command "git submodule update --init --recursive vendor/nimPNG" returned status code 1:
stdout: 
stderr: fatal: Needed a single revision
Unable to find current revision in submodule path 'vendor/nimPNG'
```
And indeed if I login onto the `windows-01` host and try that command myself I can reproduce this:
```
admin@windows-01 MINGW64 .../workspace/desktop_branches_windows_PR-7789 ((79fa2dc99...))
$ git status
HEAD detached at 79fa2dc99
nothing to commit, working tree clean

admin@windows-01 MINGW64 .../workspace/desktop_branches_windows_PR-7789 ((79fa2dc99...))
$ git submodule update --init --recursive vendor/nimPNG
fatal: Needed a single revision
Unable to find current revision in submodule path 'vendor/nimPNG'
```
And for some reason the `vendor/nimPNG` contains only one thing - a `.git` __file__, not a directory:
```
admin@windows-01 MINGW64 .../workspace/desktop_branches_windows_PR-7789 ((79fa2dc99...))
$ cd vendor/nimPNG

admin@windows-01 MINGW64 .../vendor/nimPNG (master)
$ git status
On branch master

No commits yet

nothing to commit (create/copy files and use "git add" to track)

admin@windows-01 MINGW64 .../vendor/nimPNG (master)
$ ls -la
total 13
drwxr-xr-x 1 admin 197121  0 Oct 17 12:19 ./
drwxr-xr-x 1 admin 197121  0 Oct 17 12:17 ../
-rw-r--r-- 1 admin 197121 95 Oct 17 12:19 .git 
```
Wtf is happening here?

**Comment by jakubgs** [comment]
Running:
```
admin@windows-01 MINGW64 .../vendor/nim-chronos ((bce0f87...))
$ git submodule update --init --recursive
Cloning into 'C:/Users/jenkins/workspace/desktop_branches_windows_PR-7789/vendor/nim-chronicles'...
Cloning into 'C:/Users/jenkins/workspace/desktop_branches_windows_PR-7789/vendor/nim-chronos'...
...
Submodule path '../nim-zlib/zlib/csources': checked out 'cacf7f1d4e3d44d871b605da3b647f07d718623f'
fatal: Needed a single revision
Unable to find current revision in submodule path '../nimPNG'
```
Just results in the same error, but only for that module:
```
admin@windows-01 MINGW64 .../workspace/desktop_branches_windows_PR-7789 ((79fa2dc99...))
$ ls -l vendor/nim-chronos
total 33
drwxr-xr-x 1 admin 197121     0 Oct 19 17:40 chronos/
-rw-r--r-- 1 admin 197121   376 Oct 19 17:40 chronos.nim
-rw-r--r-- 1 admin 197121   605 Oct 19 17:40 chronos.nimble
-rw-r--r-- 1 admin 197121 11573 Oct 19 17:40 LICENSE-APACHEv2
-rw-r--r-- 1 admin 197121  1130 Oct 19 17:40 LICENSE-MIT
-rw-r--r-- 1 admin 197121  1973 Oct 19 17:40 README.md
drwxr-xr-x 1 admin 197121     0 Oct 19 17:40 tests/

admin@windows-01 MINGW64 .../workspace/desktop_branches_windows_PR-7789 ((79fa2dc99...))
$ ls -l vendor/nimPNG
total 0
```
As far as I can tell this is some kind of quirk of interactions between Jenkins, Windows, and Git. Possibly a checkout that was interrupted part way during one of the builds.

**Comment by jakubgs** [comment]
And indeed if I remove the weird `.git` file which should be directory it works fine:
```
admin@windows-01 MINGW64 .../workspace/desktop_branches_windows_PR-7789 ((79fa2dc99...))
$ git submodule update --init --recursive vendor/nimPNG
fatal: Needed a single revision
Unable to find current revision in submodule path 'vendor/nimPNG'

admin@windows-01 MINGW64 .../workspace/desktop_branches_windows_PR-7789 ((79fa2dc99...))
$ rm vendor/nimPNG/.git
rm: remove regular file 'vendor/nimPNG/.git'? y

admin@windows-01 MINGW64 .../workspace/desktop_branches_windows_PR-7789 ((79fa2dc99...))
$ git submodule update --init --recursive vendor/nimPNG
remote: Enumerating objects: 1278, done.
remote: Counting objects: 100% (61/61), done.
remote: Compressing objects: 100% (35/35), done.
remote: Total 1278 (delta 26), reused 45 (delta 17), pack-reused 1217
Receiving objects: 100% (1278/1278), 20.48 MiB | 19.14 MiB/s, done.
Resolving deltas: 100% (495/495), done.
From https://github.com/jangko/nimPNG
 * [new branch]      master          -> origin/master
 * [new tag]         0.3.0           -> 0.3.0
 * [new tag]         v0.3.1          -> v0.3.1
Submodule path 'vendor/nimPNG': checked out '7ff39ec00df29b55b6de0f67d31b8d52eb5c2d8f'
```
But that doesn't actually explain why and how this this faulty file was created in the first place.

**Comment by jakubgs** [comment]
What's also worth noting is that the `cleanWs` step also failed, leaving the repo in a broken state:

image

Which seems to be due to this error:
```
org.jenkinsci.plugins.workflow.steps.MissingContextVariableException: Required context class hudson.FilePath is missing
Perhaps you forgot to surround the code with a step that provides this, such as: node, dockerNode
	at org.jenkinsci.plugins.workflow.steps.CoreStep$Execution.run(CoreStep.java:90)
	at org.jenkinsci.plugins.workflow.steps.CoreStep$Execution.run(CoreStep.java:71)
	at org.jenkinsci.plugins.workflow.steps.SynchronousNonBlockingStepExecution.lambda$start$0(SynchronousNonBlockingStepExecution.java:47)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
	at java.base/java.util.concurrent.FutureTask.run(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)
```
https://ci.status.im/job/status-desktop/job/branches/job/windows/view/change-requests/job/PR-7789/55/execution/node/15/

**Comment by anastasiyaig** [comment]
i've tested exchange of USDC to all 3 goerli networks, it works however on test net it is really slow.  but it works
lets merge it to master and tested it more there

**Review by anastasiyaig — APPROVED** [review]

**Review by stefandunca — APPROVED** [review]
Wow 👏.

**Pull request merged**